hummingbot/connector/exchange/huobi/huobi_api_order_book_data_source.py

```diff
@@ -148,93 +86,109 @@ async def _subscribe_channels(self, ws: WSAssistant):
+                snapshot_msg: OrderBookMessage = self.snapshot_message_from_exchange(
                     msg=msg,
-                    timestamp=timestamp
                 )
-                output.put_nowait(order_book_message)
-            except asyncio.CancelledError:
-                raise
-            except Exception:
-                self.logger().error("Unexpected error with WebSocket connection. Retrying after 30 seconds...",
-                                    exc_info=True)
-                await self._sleep(30.0)
-
-    async def listen_for_order_book_snapshots(self, ev_loop: asyncio.BaseEventLoop, output: asyncio.Queue):
-        while True:
-            await self._sleep(self.ORDER_BOOK_SNAPSHOT_DELAY)
-            try:
-                for trading_pair in self._trading_pairs:
-                    snapshot: Dict[str, Any] = await self.get_snapshot(trading_pair)
-                    snapshot_message: OrderBookMessage = HuobiOrderBook.snapshot_message_from_exchange(
-                        snapshot,
-                        timestamp=snapshot["tick"]["ts"],
-                        metadata={"trading_pair": trading_pair},
-                    )
-                    output.put_nowait(snapshot_message)
-                    self.logger().debug(f"Saved order book snapshot for {trading_pair}")
+                output.put_nowait(snapshot_msg)
             except asyncio.CancelledError:
                 raise
             except Exception:
-                self.logger().error("Unexpected error listening for orderbook snapshots. Retrying in 5 secs...", exc_info=True)
-                await self._sleep(5.0)
+                self.logger().exception("Unexpected error when processing public trade updates from exchange", exc_info=True)
+
+    async def listen_for_order_book_snapshots(self, ev_loop: asyncio.AbstractEventLoop, output: asyncio.Queue):
+        """
+        Suppressing call to this function as the orderbook snapshots are handled by
+        listen_for_order_book_diffs() for Huobi
+        """
+        pass
+
+    def snapshot_message_from_exchange(self,
+                                       msg: Dict[str, Any],
+                                       metadata: Optional[Dict] = None) -> OrderBookMessage:
+
+        """
+        Creates a snapshot message with the order book snapshot message
+        :param msg: the response from the exchange when requesting the order book snapshot
+        :param timestamp: the snapshot timestamp
+        :param metadata: a dictionary with extra information to add to the snapshot data
+        :return: a snapshot message with the snapshot information received from the exchange
+        """
+        if metadata:
+            msg.update(metadata)
+        msg_ts = int(round(msg["tick"]["ts"] / 1e3))
+        msg_channel = msg["ch"]
+        content = {
+            "update_id": msg["tick"]["ts"],
+            "bids": msg["tick"]["bids"],
+            "asks": msg["tick"]["asks"]
```

Comment:
We should prepare the code for cases when the exchange sends an update with only bids or only asks.
```suggestion
            "bids": msg["tick"]get("bids", []),
            "asks": msg["tick"].get("asks", [])
```